### DISCUSSION TITLE
Shrink and replace the original file

### DISCUSSION THREAD

**Discussion by MorneArin** [ZEISS/czicompress#56] [CZI Shrink Questions]
### Question

Hi all,
is it possible to shrink and replace the original file automatically, without the need to create a separate folder?
Thanks

**Comment by DaveyJonesBitPail** [comment]
No, this is currently not possible.

If the input directory and output directory are defined as the same location the application will also avoid overwriting existing files and will log an error for the operation.

From a motivation standpoint we wanted to make it very difficult to accidentally corrupt existing files during overwrite which can occur for a myriad of reasons.

If this functionality is necessary for your use case then we could create a feature request.

**Comment by m-ringler** [comment]
You can use the command line tool czicompress for that. It should be easy to adapt one of the scripts in the [Readme](https://github.com/ZEISS/czicompress/blob/main/czicompress/README.md) to move the output file to the location of the input file after successful (!) compression. If you're not that much into scripting, Chat GPT can probably help with that.
But as @DaveyJonesBitPail has said, it's easy to shoot yourself in the foot if you're not careful.
